### PR TITLE
E2E: Base layer tests

### PR DESCRIPTION
**Pull request by 4ydan**
3 scenarios for testing the base layer.

- add background
- rotate background
- scale background

Also adds mypy config file for a bit stricter checks.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by 4ydan** [comment]
jenkins build please

**Comment by 4ydan** [comment]
jenkins build please

File: e2e/features/base_layer.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Feature: PermaplanT base layer
+    As a user I want to be able to have a base layer
+
+  Background:
+    Given I am on the SUTBaseLayer map page and I have selected the base layer
+
+    Scenario: Successfully select a background image
+        When I select the background image Birdie.jpg
+        Then /Photos/Birdie.jpg stays even when I leave SUTBaseLayer and come back later
```
Comment (markus2330) [inline e2e/features/base_layer.feature:9]: This will change today, as we will discuss in the meeting.

File: e2e/pages/maps/planting.py [hunk 2]
```diff
@@ -22,6 +23,13 @@ def __init__(self, page: Page):
         self.canvas = self.page.get_by_test_id("canvas")
         self.undo_button = self.page.get_by_test_id("undo-button")
         self.redo_button = self.page.get_by_test_id("redo-button")
+        # Base layer
+        self.background_select = self.page.get_by_test_id("baseBackgroundSelect")
+        self.background_button = page.get_by_role("button", name="Choose an image")
+        self.rotation_input = self.page.get_by_test_id("rotation-input")
+        self.scale_input = self.page.get_by_test_id("scale-input")
```
Comment (markus2330) [inline e2e/pages/maps/planting.py:30]: This is actually totally unrelated to Planting, it should be in a separate class/file?

File: e2e/steps/test_base_layer.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+from e2e.pages.maps.create import MapCreatePage
+from e2e.pages.maps.planting import MapPlantingPage
+from e2e.pages.maps.management import MapManagementPage
+from pytest_bdd import scenarios, given, when, then, parsers
+
+scenarios("features/base_layer.feature")
+
+
+@given(parsers.parse("I am on the {name} map page and I have selected the base layer"))
+def on_planting_screen(
```
Comment (markus2330) [inline e2e/steps/test_base_layer.py:13]: planting screen != map editor

**Review by markus2330 — APPROVED** [review]
LGTM, refactor with changing of the Image.
